app/src/main/java/com/github/multimatum_team/multimatum/activity/AddDeadlineActivity.kt

```diff
@@ -414,8 +464,8 @@ class AddDeadlineActivity : AppCompatActivity() {
     // Recuperate the information on which notification must be set before returning it in an array
     private fun retrieveNotificationsTimes(): ArrayList<Long> {
         val res = ArrayList<Long>()
-        for (i in 0 until checkBoxIdTime.count()) {
-            if (notificationSelected[i]) res.add(checkBoxIdTime[nameNotifications[i]]!!)
+        for (i in 0 until timeNotifications.count()) {
+            if (notificationSelected[i]) res.add(timeNotifications[i])
         }
         return res
```

Comment:
```suggestion
        return timeNotifications.filterIndexed { idx, _ -> notificationsSelected[idx] }
```
(not 100% sure of the syntax but should be very close to that)

Reply:
I didn't know that, thanks !